test/integration/targets/conditionals/tasks/main.yml

```diff
@@ -307,3 +307,29 @@
 - name: test complex templated condition
   debug: msg="it works"
   when: vars_file_var in things1|union([vars_file_var])
+
+- name: test dict with invalid key does not run with conditional is defined
+  vars:
+    mydict:
+      a: foo
+      b: bar
+  debug: var=mydict['c']
+  when: mydict['c'] is defined
+  register: result
+
+- name: assert the task was skipped
+  assert:
+    that:
+    - "result.skipped == true"
+
+- name: test list with invalid element does not run with conditional is defined
+  vars:
+    mylist: []
+  debug: var=mylist[0]
+  when: mylist[0] is defined
+  register: result
+
+- name: assert the task was skipped
+  assert:
+    that:
+    - "result.skipped == true"
```

Comment:
Removed my fix since a better one was already merged.
I added tests for an is undefined conditional. These shouldn't fail with debug (as was happening), but will output something like
```
TASK [conditionals : test list with invalid element is undefined] ***********************************************************************************************************************************************************************************************
ok: [testhost] => {
    "failed": false,
    "mylist[0]": "VARIABLE IS NOT DEFINED!"
}
```